lib/iris/coords.py

```diff
@@ -1219,7 +1219,7 @@ def serialize(x):
             # Calculate the bounds and points along the right dims
             bounds = al.stack([item.min(axis=dims_to_collapse),
                                item.max(axis=dims_to_collapse)]).T
-            points = al.array(bounds.sum(axis=-1) * 0.5, dtype=self.dtype)
+            points = item.mean(axis=dims_to_collapse, dtype=self.dtype)
```

Comment:
There was a [question at #3008](https://github.com/SciTools/iris/pull/3008#issuecomment-385983949) about whether multiple collapses are commutative.  It looks like by averaging `item` here, which is the `bounds` of the input coord where available, we do  lose commutativity.  E.g.

```python
mycoord = iris.coords.AuxCoord([[1, 2], [10, 11], [100, 101]], long_name='foo')
print(mycoord.collapsed())
print(mycoord.collapsed(0).collapsed())
print(mycoord.collapsed(1).collapsed())
```

gives:
```
AuxCoord(array([37]), bounds=array([[  1, 101]]), standard_name=None, units=Unit('1'), long_name='foo')
AuxCoord(array([51]), bounds=array([[  1, 101]]), standard_name=None, units=Unit('1'), long_name='foo')
AuxCoord(array([37]), bounds=array([[  1, 101]]), standard_name=None, units=Unit('1'), long_name='foo')
```
Is this the intended behaviour, or should the new points be averaged from the input points?